enhancements/ocp-coreos-layering/ocp-coreos-layering.md

```diff
@@ -0,0 +1,741 @@
+
+##### Handling extensions
+
+We need to preserve support for [extensions](https://github.com/openshift/enhancements/blob/master/enhancements/rhcos/extensions.md). For example, `kernel-rt` support is key to many OpenShift use cases.
+
+Extensions move to a `rhel-coreos-extensions` container that has RPMs and will be built part of release payload.
+
+To apply an extension, MCO will pull in RPMs from `rhel-coreos-extensions` container image from the release payload and apply the required extensions to the targeted nodes like we do currently.
+
+Cluster admin shouldn't see any change in the behavior. Interface for applying `kernelType` and supported `extensions` will remain same, the way it is done today via MachineConfig. See https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfiguration.md#rhcos-extensions .
+
+##### Kernel Arguments
+
+For now, updating kernel arguments will continue to happen via the MCD on each node via executing `rpm-ostree kargs` as it does today. See also https://github.com/ostreedev/ostree/issues/479
+
+##### Drain and reboot
+
+No changes in drain and reboot behavior, MCO will continue to perform drain and reboots like it does for existing releases.
+
+##### Telemetry
+
+We will be collecting telemetry data on how many clusters are using custom `osImageURL`. This data will help us understand how frequently layering feature is getting used as well as help us improve our future layering work.
+
+### Risks and Mitigations
+
+We're introducing a new format of shipping RHEL CoreOS image and allowing users to create their own custom image to override the default os on nodes.
+
+This brings in certain risk but hopefully with good documentation, cluster admins would know what they are doing and tests it before applying custom Image to cluster.
+
+### Drawbacks
+
+If it turns out that e.g. upgrading existing RHCOS systems in place is difficult, that will be a problem.
+
+## Design Details
+
+### Test Plan
+
+The FCOS / RHCOS base OCI containers will be published automatically by their respective CI pipelines. These pipelines already have a release promotion mechanism built into them. This means that early development versions of these containers are available (e.g. `:testing-devel` tags), hence other dependent systems, such as the MCO, can use them.
+
+For example, the MCO’s tests can use those containers to get a signal on potentially breaking changes to rpm-ostree as well as the base container images.
+
+MCO-specific tests will be run during each PR as part of the MCOs presubmits. Also, make sure that we're doing node scaling post-upgrade in some e2e tests.
+
+### Graduation Criteria
+
+ The goal for 4.12 is to have a supported path for users to apply hotfix (GA) into base RHEL CoreOS and ability to try out 3rd party software (Tech Preview).
+
+#### Dev Preview -> Tech Preview
```

Comment:
Can we clean up this section since we are going straight to GA for hotfix?

Reply:
done, hopefully this looks good now.